ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/query/json/ContractStateVaultJsonFactory.java

```diff
@@ -0,0 +1,67 @@
+package net.corda.v5.ledger.utxo.query.json;
+
+import net.corda.v5.application.marshalling.JsonMarshallingService;
+import net.corda.v5.ledger.utxo.ContractState;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * The interface that needs to be implemented in order to create a JSON representation of a state type. Implementing
+ * this interface will define how a state can be represented as a JSON string.
+ * <p>
+ * Example usage:
+ * <ul>
+ * <li>Kotlin:<pre>{@code
+ * class TestUtxoStateVaultJsonFactory: ContractStateVaultJsonFactory<TestUtxoState> {
+ *
+ *     override val stateType: Class<TestUtxoState> = TestUtxoState::class.java
+ *
+ *     override fun append(state: TestUtxoState, jsonMarshallingService: JsonMarshallingService): String {
+ *         return jsonMarshallingService.format(TestUtxoStatePojo(state.testField))
+ *     }
+ * }
+ *
+ * data class TestUtxoStatePojo(val testField: String)
```

Comment:
Can you make the `data class` the same as the java example, e.g. a private class in the same position in the file? Just so it makes the examples more equivalent even though they both fine as standalone examples.